lib/utils/sort-locales.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Given a list of locales, this utility will sort them by `fallback` order, i.e. the locales without fallbacks first,
+ * then the locales having them as fallbacks, and then recursively.
+ */
+export default function sortLocales (locales) {
+  const localeByFallback = {}
+
+  locales.forEach(locale => {
+    if (locale.fallbackCode === null) {
+      locale.fallbackCode = undefined
+    }
+
+    if (!localeByFallback[locale.fallbackCode]) {
+      localeByFallback[locale.fallbackCode] = []
+    }
+
+    localeByFallback[locale.fallbackCode].push(locale)
+  })
+
+  return sortByFallbackKey(undefined, localeByFallback)
+}
+
+function sortByFallbackKey (key, localeByFallback) {
```

Comment:
Given the parameter `key` can be undefined, `sortByFallbackKey` should probably have the `key` parameter specified as the last parameter.

I.e. 
```
function sortByFallbackKey(localeByFallback, key)
```

and invocation on line 20, would look like this
```
sortByFallbackKey(localeByFallback)
```

`key` would still be undefined.